modules/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobContainerRetriesTests.java

```diff
@@ -472,4 +474,32 @@ public void testRetryFromSecondaryLocationPolicies() throws Exception {
             assertThat(failedGetCalls.get(), equalTo(1));
         }
     }
+
+    public void testPrematureClosedConnectionDoesNotInterruptBackingThread() throws Exception {
```

Comment:
could you please point on the place where you simulate thread.interrupt()? is it missed or interruption is implicit somehow?

Reply:
The interrupt comes from the future being cancelled due to the premature connection close set up in the `httpServer.createContext` line - if you undo the fix (e.g. apply the following patch) then the test fails:

```diff
diff --git a/modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/executors/ReactorScheduledExecutorService.java b/modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/executors/ReactorScheduledExecutorService.java
index f621cfe3e979..4c2c378acb12 100644
--- a/modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/executors/ReactorScheduledExecutorService.java
+++ b/modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/executors/ReactorScheduledExecutorService.java
@@ -202,7 +202,7 @@ public class ReactorScheduledExecutorService extends AbstractExecutorService imp
         @Override
         public boolean cancel(boolean mayInterruptIfRunning) {
             // Ensure that the thread is never interrupted
-            return delegate.cancel(false);
+            return delegate.cancel(mayInterruptIfRunning);
         }

         @Override
```